src/index.js

```diff
@@ -314,11 +314,11 @@ class PubsubBaseProtocol extends EventEmitter {
    * For example, a Floodsub implementation might simply publish each message to each topic for every peer
    * @abstract
    * @param {Array<string>|string} topics
-   * @param {Array<any>|any} messages
-   * @returns {Promise}
+   * @param {Buffer | string} message
```

Comment:
```suggestion
   * @param {Buffer} message
```
Considering that we are breaking this, I would go with only Buffer, to be the same as https://github.com/libp2p/js-libp2p/blob/master/doc/API.md#pubsubpublish 
What do you think? Reducing the scope means less code to maintain and less edge cases that might appear

Reply:
:thumbsup: I just set this bc that is what the protobuf encoder will accept.

Reply:
sounds fair to me